### ISSUE TITLE
<unnamed> (81, 19): parse expected comma after load's type

### ISSUE THREAD

**Issue by AlexKurek** [numba/numba#6607]
### Reporting a bug

Im getting an error like pasted below while trying to run the most basic example taken from official WWW:
https://developer.nvidia.com/blog/numba-python-cuda-acceleration/

```
import numpy as np
from numba import vectorize

@vectorize(['float32(float32, float32)'], target='cuda')
def Add(a, b):
  return a + b

N = 100000
A = np.ones(N, dtype=np.float32)
B = np.ones(A.shape, dtype=A.dtype)
C = np.empty_like(A, dtype=A.dtype)

C = Add(A, B)
```
Ubuntu 18.04.5, Numpy 1.19.4, Numba 0.52.0, kernel 5.4.58-050458-generic, CUDA 11.2.0, NVIDIA driver 460.27.04, GeForce GTX 1660

> Traceback (most recent call last):
>   File "GPUtest.py", line 13, in <module>
>     C = Add(A, B)
>   File "/usr/local/lib/python3.6/dist-packages/numba/cuda/dispatcher.py", line 37, in __call__
>     return CUDAUFuncMechanism.call(self.functions, args, kws)
>   File "/usr/local/lib/python3.6/dist-packages/numba/np/ufunc/deviceufunc.py", line 300, in call
>     cr.launch(func, shape[0], stream, devarys)
>   File "/usr/local/lib/python3.6/dist-packages/numba/cuda/dispatcher.py", line 148, in launch
>     func.forall(count, stream=stream)(*args)
>   File "/usr/local/lib/python3.6/dist-packages/numba/cuda/compiler.py", line 372, in __call__
>     kernel = self.kernel.specialize(*args)
>   File "/usr/local/lib/python3.6/dist-packages/numba/cuda/compiler.py", line 899, in specialize
>     self._bind, targetoptions)
>   File "/usr/local/lib/python3.6/dist-packages/numba/cuda/compiler.py", line 811, in __init__
>     self.compile(sigs[0])
>   File "/usr/local/lib/python3.6/dist-packages/numba/cuda/compiler.py", line 952, in compile
>     kernel.bind()
>   File "/usr/local/lib/python3.6/dist-packages/numba/cuda/compiler.py", line 576, in bind
>     self._func.get()
>   File "/usr/local/lib/python3.6/dist-packages/numba/cuda/compiler.py", line 446, in get
>     ptx = self.ptx.get()
>   File "/usr/local/lib/python3.6/dist-packages/numba/cuda/compiler.py", line 416, in get
>     **self._extra_options)
>   File "/usr/local/lib/python3.6/dist-packages/numba/cuda/cudadrv/nvvm.py", line 548, in llvm_to_ptx
>     ptx = cu.compile(**opts)
>   File "/usr/local/lib/python3.6/dist-packages/numba/cuda/cudadrv/nvvm.py", line 236, in compile
>     self._try_error(err, 'Failed to compile\n')
>   File "/usr/local/lib/python3.6/dist-packages/numba/cuda/cudadrv/nvvm.py", line 254, in _try_error
>     self.driver.check_error(err, "%s\n%s" % (msg, self.get_log()))
>   File "/usr/local/lib/python3.6/dist-packages/numba/cuda/cudadrv/nvvm.py", line 144, in check_error
>     raise exc
> numba.cuda.cudadrv.error.NvvmError: Failed to compile
> 
> <unnamed> (81, 19): parse expected comma after load's type
> NVVM_ERROR_COMPILATION

Is some default behavior changed and broke the decorator?

**Comment by AlyShmahell** [comment]
I am facing the same issue on Windows 10, with:
- GTX 1050 Ti
- Nvidia Driver Version 460.89
- cuda_11.2.0_460.89_win10
- numba==0.52.0
- llvmlite==0.35.0

**Comment by AlexKurek** [comment]
@AlyShmahell Have you tried to downgrade CUDA?

**Comment by esc** [comment]
@AlexKurek thank you for reporting this. It will need some investigation to figure out what is going on here.

**Comment by stuartarchibald** [comment]
CUDA 11.2 isn't currently supported by Numba. Downgrading should let it work.

**Comment by AlexKurek** [comment]
@stuartarchibald What is the latest CUDA we can use currently?

**Comment by stuartarchibald** [comment]
> @stuartarchibald What is the latest CUDA we can use currently?

@AlexKurek I think it's anything supported and <11.2, so 11.1.x would be the last compatible. We are aware of the issues and there's work ongoing to provide 11.2+ support.

**Comment by AlyShmahell** [comment]
@AlexKurek yes it worked after downgrading to the following:
- cuda version 10.1
- llvmlite==0.31
- numba==0.48

**Comment by gbliao** [comment]
I experienced this issue with CUDA 11.6. The issue was resolved by using cuda 11.1 instead.
